### PR TITLE
[ui] Migrate project settings to React

### PR DESCRIPTION
**Pull request by dcramer**
TODO:
- [x] Transfer Project
- [x] Remove Project
- [x] Org settings that can't be overridden

### CODE REVIEW

**Comment by MaxBittker** [comment]
nice 👌

**Comment by dcramer** [comment]
dont think this is done yet but i rebased/added verify_ssl

File: src/sentry/api/endpoints/project_details.py [hunk 1]
```diff
@@ -239,36 +316,74 @@ def put(self, request, project):
                 project=project
             )
 
+        # TODO(dcramer): rewrite options to use standard API config
         if has_project_write:
             options = request.DATA.get('options', {})
             if 'sentry:origins' in options:
```
Comment (dcramer) [inline src/sentry/api/endpoints/project_details.py]: goal will be to remove these bits down the road

**Comment by dcramer** [comment]
@getsentry/workflow this was some early work i did on settings -- we may not need it, but we can close it out/do whatever with it

File: src/sentry/api/endpoints/project_details.py [hunk 2]
```diff
@@ -251,12 +288,40 @@ def put(self, request, project):
         if result.get('digestsMaxDelay'):
             project.update_option(
                 'digests:mail:maximum_delay', result['digestsMaxDelay'])
-        if result.get('subjectPrefix'):
+        if result.get('subjectPrefix') is not None:
```
Comment (billyvg) [inline src/sentry/api/endpoints/project_details.py]: Previously you wouldn't be able to remove a value, unsure if we want user to be able to delete some of these keys below.

File: src/sentry/api/serializers/models/project.py [hunk 3]
```diff
@@ -325,11 +317,28 @@ def serialize(self, obj, attrs, user):
                     'digests:mail:maximum_delay',
                     digests.maximum_delay,
                 ),
-                'subjectPrefix':
-                attrs['options'].get('mail:subject_prefix'),
+                'allowedDomains':
+                attrs['options'].get(
+                    'sentry:origins', ['*']),
+                'resolveAge':
```
Comment (billyvg) [inline src/sentry/api/serializers/models/project.py]: This was a dcramer change, probably for frontend consumption reasons....
it's unclear when things should be in the `options` hash i.e. `digestsMinDelay`

File: src/sentry/web/urls.py [hunk 4]
```diff
@@ -416,7 +415,7 @@ def init_all_applications():
     # Settings - Projects
     url(
         r'^(?P<organization_slug>[\w_-]+)/(?P<project_slug>[\w_-]+)/settings/$',
-        ProjectSettingsView.as_view(),
+        react_page_view,
```
Comment (billyvg) [inline src/sentry/web/urls.py]: It seems we have two ways of doing this, link to absolute uri or define a route's view to `react_page_view` (and thus being able to keep using django's named routes). What's the preferred method here?

**Comment by billyvg** [comment]
Rebased w/ master, fixed some fields that weren't working. Manually tested all fields.

TODO:
- [ ] Transfer Project
- [ ] Remove Project
- [ ] Org settings that can't be overridden

File: src/sentry/models/project.py [hunk 5]
```diff
@@ -314,3 +315,11 @@ def add_team(self, team):
             return False
         else:
             return True
+
+    def get_security_token(self):
+        # TODO(dcramer): this update should happen within a lock
```
Comment (macqueen) [inline src/sentry/models/project.py]: @dcramer should this happen before this gets merged?

File: src/sentry/api/serializers/rest_framework/origin.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import
+
+from rest_framework import serializers
+
+from sentry.utils.http import parse_uri_match
+
+
+class OriginField(serializers.CharField):
```
Comment (macqueen) [inline src/sentry/api/serializers/rest_framework/origin.py]: this seems like something it'd be good to have tests for

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Review by macqueen — APPROVED** [review]

**Comment by billyvg** [comment]
I think we're going to need to wait for https://github.com/getsentry/sentry/pull/6727 since remove/transfer project require `sudo`

**Pull request merged**